components/clp-py-utils/clp_py_utils/profiling_utils.py

```
@@ -0,0 +1,256 @@
+        Can use dot notation for attributes, e.g., "task.id".
+    :return: Decorated function with profiling capabilities.
+    """
+
+    def decorator(func: F) -> F:
+        name = section_name or func.__name__
+        is_async = inspect.iscoroutinefunction(func)
+
+        if is_async:
+
+            @functools.wraps(func)
+            async def async_wrapper(*args, **kwargs):
```

Ruff (0.14.1)

Missing return type annotation for private function `async_wrapper`

(ANN202)